src/caustics/light/point_source.py

```diff
@@ -0,0 +1,163 @@
+# mypy: disable-error-code="operator,union-attr"
+from typing import Optional, Union, Annotated
+
+from torch import Tensor
+
+from .base import Source, NameType
+from ..parametrized import unpack
+from ..packed import Packed
+from . import func
+
+__all__ = ("PointSource",)
+
+
+class PointSource(Source):
+    """
+    `Point` is a subclass of the abstract class `Source`.
+    It represents a point source in a gravitational lensing system.
+
+    The Point profile is meant to describe individual light sources
+    whose angular extent is much less than the Einstein radius of the lens.
+
+    Attributes
+    -----------
+    x0: Optional[Tensor]
+        The x-coordinate of the Point source's center.
+
+        *Unit: arcsec*
+
+    y0: Optional[Tensor]
+        The y-coordinate of the Point source's center.
+
+        *Unit: arcsec*
+
+    Ie: Optional[Tensor]
+        The intensity at the point.
+
+        *Unit: flux*
+
+    s: float
+        A small constant for numerical stability.
+
+        *Unit: arcsec*
+
+
+
+    """
+
+    def __init__(
+        self,
+        x0: Annotated[
+            Optional[Union[Tensor, float]],
+            "The x-coordinate of the Sersic source's center",
+            True,
+        ] = None,
+        y0: Annotated[
+            Optional[Union[Tensor, float]],
+            "The y-coordinate of the Sersic source's center",
+            True,
+        ] = None,
+        theta_s: Annotated[
+            Optional[Union[Tensor, float]],
+            "The radius of the point source",
+            True,
+        ] = None,
+        Ie: Annotated[
+            Optional[Union[Tensor, float]],
+            "The intensity at the effective radius",
+            True,
+        ] = None,
+        gamma: Annotated[
+            Optional[Union[Tensor, float]],
+            "The linear limb darkening coefficient",
+            True,
+        ] = None,
+        s: Annotated[float, "A small constant for numerical stability"] = 0.0,
```

Comment:
I don't think `s` is needed here for numerical stability, so it can just be left out